View/Pages/Matching.xaml.cs

```diff
@@ -1,9 +1,45 @@
 ﻿namespace View.Pages;
+using Model;
+using Controller;
+using System.Linq;
 
 public partial class Matching : ContentPage
 {
+    readonly static User user = Auth.getUser();
+
+    //Dit wordt straks gekoppeld aan de lijst met potentiele matches die uit het algoritme komen (Input filter).
+    public List<User> Users => dummydb.Users;
+
+    //Een lijst met potentiele matches die zijn gefilterd op plaats/dorp.
+    public List<User> FilterPlace { get; set; } = new List<User>();
+
+    //De uiteindelijke lijst potentiele matches die zijn gefilterd op plaats/dorp en leeftijd (Output filter).
+    public List<User> FilteredPotentionalMatches => FilterPlace.Where(x => x.Age >= user.MinimalAge).Where(x => x.Age <= user.MaximalAge).ToList();
+
     public Matching()
     {
         InitializeComponent();
+        FilterCitys();
+        BindingContext = this;
+    }
+
+    /// <summary>
+    /// Filters potential matches that have selected the same citys
+    /// </summary>
+    public void FilterCitys()
+    {
+        foreach (User userMatch in Users)
```

Comment:
kan deze hele ellende niet worden vervangen door iets van een `user.Cities.Intersect(user.Cities)` zie [dit](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.intersect?view=net-6.0)

Reply:
Het zou kunnen dat we hiervoor dan ook een kleine custom Equals methode moeten schrijven die checkt of de namen gelijk zijn

Reply:
Wat schiet je er mee op?

Reply:
Betere code?

Reply:
en hoogstwaarschijnlijk ook efficienter